classes/patterns/class-base-pattern.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Base pattern class.
+ *
+ * @package P4GBKS
+ */
+
+namespace P4GBKS\Patterns;
+
+/**
+ * Class Base_Pattern
+ *
+ * @package P4GBKS\Patterns
+ */
+abstract class Base_Pattern {
+
+	public const NAMESPACE = 'planet4-patterns';
+
+	/**
+	 * Returns the pattern name with its namespace prefix, e.g.: planet4-patterns/side-image-with-text-and-cta.
+	 */
+	public static function get_full_pattern_name() {
+		return static::NAMESPACE . '/' . static::PATTERN_NAME;
```

Comment:
The problem with constants is that you can't require them to be present with an abstract class. Abstract functions on the other hand will need to have an implementation in any extending class, or the code won't even run (or pass CI tests).

In other words, if we create 2 abstract functions (`get_name` and `get_attributes`) is a way of saying ""Every pattern needs to have a name and an array of attributes".